### ISSUE TITLE
I have some question about the input and output

### ISSUE THREAD

**Issue by chendiva** [kdgutier/esrnn_torch#9]
Hi there, 
    So I am now using a time series data which only have two columns- Date and Price.  So I am wondering if I can use this algorithm in this situation, and let the algorithm train the model only on price, and predict the price in the future. In other words,
I am wondering if this model can separate my data automatically, so that I will not need to separate by "lag" myself. Thank you for your help!

**Comment by kdgutier** [comment]
Hi Chendiva,
Try to parse your data as mentioned in the README file (
https://github.com/kdgutier/esrnn_torch) with the price data in the y_df,
try to add a simple constant in the X_df to use if you don't have any
exogenous variable. The algorithm needs you to plug with the same column
names as specified by the README dataframes, the lag variables are
calculated within the algorithm. Good luck.

**Comment by chendiva** [comment]
Hi,
   Will it affect the forecasting result when I add the exogenous variable?

**Comment by kdgutier** [comment]
I recommend you to always have benchmarks to test complex models as the
ESRNN, in our case we included the OWA metric in the validation set to
compare the relative performance vs the Naive2 model as done in the M4
competition. Take extra care of the learning rate hyperparameters when
tuning your model.

**Comment by chendiva** [comment]
Sorry , I am still confused, if I add the exogenous variable as you recommend, will it affect the result? Is the x variable in your example added by you? Or this x is originally included in the dataset and use for forecasting?

**Comment by kdgutier** [comment]
I would recommend to answer the question empirically (with the OWA metric),
try the model and see if the performance remains acceptable.

**Comment by chendiva** [comment]
Hi there,
   I see that the y_test_df has s column of y_hat_naive_2. How do you get this column?

**Comment by chendiva** [comment]
This also give me NaN for my y_hat

**Comment by kdgutier** [comment]
Hi chendiva,
I checked the bug, the ESRNN produces correct outputs that fail to merge in
the predict method to the X_test_df if the frequency of the dataset is not
correctly specified.
For instance in the M3 dataset the frequency seems to be 'MS' for dates of
the beginning of the month. In the bug reported before they were using 'M'
frequency for dates at the end of the month. Let me know if this solves the
problem.

**Comment by chendiva** [comment]
Hi, how can I decide the frequency then?

**Comment by kdgutier** [comment]
When you instantiate the ESRNN model
model = ESRNN(params,...,frequency=‘MS’)

**Comment by chendiva** [comment]
I actually use my dataset, not the M3 now. My dataset is daily base. so I set the frequency = 'D', but I then got the error like this:
![image](https://user-images.githubusercontent.com/22489898/88876633-b3145880-d1f1-11ea-9e72-9cb9161432ee.png)

**Comment by kdgutier** [comment]
That is a protection for the network that aims to protect the model from
nan values. Clean nans from the data before using the ESRNN.

**Comment by chendiva** [comment]
Yes, I actually check the dataframe with this command: df.isnull().values.any(), which returns me False. But I still get the above result

**Comment by kdgutier** [comment]
Have you tried printing those unique_ids?
Also isnan function?

**Comment by chendiva** [comment]
I actually got this after using the function you mentioned:
![image](https://user-images.githubusercontent.com/22489898/88877408-63369100-d1f3-11ea-9fff-69b4875d9096.png)

**Comment by kdgutier** [comment]
You are treating the unique_ids as a numeric variable. I suggest to check
the README markdown of the github in which the input dataframes for the
model are explained with detail.

**Comment by chendiva** [comment]
However, the id that I am using is copied from the m4 example, which is weird?

**Comment by Yu-1245** [comment]
Hi, there, I got the same problem with yours, have you solved it? I tried to slice the m4 data provided from the prepare_m4_data function, and found out that even I make sure the identifier in the training set and testing set are the same, it still generated NaN for the evaluation methods and the predictions, which was weird.

**Comment by FedericoGarza** [comment]
Hi!

I think [this answer](https://github.com/kdgutier/esrnn_torch/issues/7#issuecomment-667239239) could be useful.

**Comment by Yu-1245** [comment]
Hi, 
I saw the answer you. I have checked my dataset and make the changed you mentioned, but it still generate NaN for me.
@FedericoGarza

**Comment by chendiva** [comment]
> Hi, there, I got the same problem with yours, have you solved it? I tried to slice the m4 data provided from the prepare_m4_data function, and found out that even I make sure the identifier in the training set and testing set are the same, it still generated NaN for the evaluation methods and the predictions, which was weird.

No, I haven't solved the problem yet, even I tried his method.

**Comment by kdgutier** [comment]
Have you solved the issue Worben?
